internal/controllers/gateway/route_utils_test.go

```diff
@@ -208,6 +209,10 @@ func Test_getSupportedGatewayForRoute(t *testing.T) {
 			{
 				name: "basic HTTPRoute gets accepted",
```

Comment:
Could we split the `Test_getSupportedGatewayForRoute` into functions for different types of routes? This test function is near 1K lines now.

Reply:
This already uses subtests for particular route types (`HTTPRoute`, `TCPRoute`) since generics do not allow to reuse a struct in a generic way we'd want here.

Additionally I suggest to use more of the the builder helper funcs in subsequent PRs ( I already added usage of Listener builder in: `Add Gateway API listener builder` #3183).

WDYT?

Reply:
I think it is OK if doing the things altogether will make the PR too large. At least these are unit tests.